File: 27.py
# -*- coding:utf-8 -*-
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None
class Solution:
    # 返回镜像树的根节点
    def core(self,pNode:TreeNode):
        if(pNode==None):
            return
        temp = pNode.left
        pNode.left = pNode.right
        pNode.right = temp
        self.core(pNode.right)
        self.core(pNode.left)
    def Mirror(self, root:TreeNode):
        self.core(root)
        return root
# write code here

File: 27.cpp
//
// Created by zb198 on 2019/9/24.
//

struct TreeNode {
	int val;
	struct TreeNode *left;
	struct TreeNode *right;
	TreeNode(int x) :
			val(x), left(nullptr), right(nullptr) {
	}
};
class Solution {
private:
    void core(TreeNode* pNode){
        if(pNode== nullptr){
            return;
        }
        TreeNode* temp = pNode->left;
        pNode->left = pNode->right;
        pNode->right = temp;
        core(pNode->right);
        core(pNode->left);
    }
public:
    void Mirror(TreeNode *pRoot) {
        core(pRoot);
        return;pRoot;
    }
};

